src/libexpr/nixexpr.hh

```diff
@@ -79,16 +79,28 @@ struct Expr
 {
     virtual ~Expr() { };
     virtual void show(std::ostream & str) const;
+    virtual void showAsAterm(std::ostream & str) const;
+    virtual void showAsJson(std::ostream & str) const;
```

Comment:
This should be
```
virtual nlohmann::json to_json() const;
```
or something like that.

Reply:
`nlohmann::json` sounds like overhead .. intermediary data structure + unnecessary pretty printing

Reply:
that [dump](https://github.com/nlohmann/json/blob/6f551930e5c7ef397056de121c0da82f77573cca/include/nlohmann/detail/output/serializer.hpp#L96) function is similar to my implementation of `showAsJson`, also using recursion, not iteration